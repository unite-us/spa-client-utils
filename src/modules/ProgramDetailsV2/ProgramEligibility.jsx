import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Divider } from '@unite-us/ui';
import DraftEditorField from 'modules/form/DraftEditorField';
import ExpandableContainer from 'modules/ExpandableContainer';
import LabelTextInline from 'modules/Shared/LabelTextInline';


const ProgramEligibility = (props) => {
  const {
    className,
    program: {
      accessibility_options,
      cities,
      counties,
      delivery_options,
      description,
      eligibility_text,
      languages,
      payment_options,
      population_specializations,
      population_restrictions,
      states,
      transportation_options,
    },
    sectionEmpty,
    showLess,
    showMore,
  } = props;

  return (
    <div className={classNames('program-description__eligibility', className)} >
      <div className={classNames({ hidden: sectionEmpty.eligibility })}>
        <ExpandableContainer
          collapsedHeight="70px"
          id="test-expandable"
          controlled
          isOpen={showMore}
          hideToggle
          showGradient={showLess}
        >
          <h4 className="mb-half">Eligibility</h4>
          <DraftEditorField
            className="pb-one"
            hideLabel
            id="program-eligibility_text"
            label="Eligibility"
            readOnly
            value={eligibility_text || ''}
          />
        </ExpandableContainer>
      </div>

      {/* Adding additional expandable when eligibility DNE */}

      {
        sectionEmpty.eligibility && description && showLess ? (
          <ExpandableContainer
            collapsedHeight="70px"
            id="test-expandable"
            controlled
            isOpen={showMore}
            hideToggle
            showGradient={showLess}
          >
            <h4 className="mb-one">Description</h4>
            <DraftEditorField
              className="mb-two"
              label="Description"
              hideLabel
              id="program-description-text"
              value={description}
              readOnly
            />
          </ExpandableContainer>
        ) : null
      }

      {
        !sectionEmpty.geography && showMore ? (
          <div>
            {!sectionEmpty.eligibility ?
              (<Divider className="program-details__divider" />) : null}

            <h4 className="mb-one">Client must live in the following...</h4>
            <LabelTextInline
              className="mb-one"
              label="States"
              content={states}
            />
            <LabelTextInline
              className="mb-one"
              label="Counties"
              content={counties}
            />
            <LabelTextInline
              className="mb-one"
              label="Cites"
              content={cities}
            />
          </div>
        ) : null
      }

      {
        showMore && (
          <div>
            {!sectionEmpty.geography || !sectionEmpty.eligibility ?
              (<Divider className="program-details__divider" />) : null}
            <LabelTextInline
              className="mb-one"
              label="Populations restricted to"
              content={population_restrictions}
            />

            <LabelTextInline
              className="mb-one"
              label="Populations catered to"
              content={population_specializations}
            />

            <LabelTextInline
              className="mb-one"
              label="Languages"
              content={languages}
            />

            <LabelTextInline
              className="mb-one"
              label="Transportation"
              content={transportation_options}
            />

            <LabelTextInline
              className="mb-one"
              label="Accessibility"
              content={accessibility_options}
            />

            <LabelTextInline
              className="mb-one"
              label="Primary Method of Service Delivery"
              content={delivery_options}
            />

            <LabelTextInline
              className="mb-one"
              label="Payment options"
              content={payment_options}
            />
          </div>
        )
      }
    </div>
  );
};

ProgramEligibility.propTypes = {
  className: PropTypes.string,
  showMore: PropTypes.bool,
  showLess: PropTypes.bool,
  program: PropTypes.object.isRequired,
  sectionEmpty: PropTypes.object,
};

ProgramEligibility.defaultProps = {
  className: '',
  sectionEmpty: {
    eligibility: false,
    geography: false,
  },
  showLess: false,
  showMore: false,
};

export default ProgramEligibility;
