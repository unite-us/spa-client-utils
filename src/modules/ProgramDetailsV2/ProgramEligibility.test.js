import React from 'react';
import renderer from 'react-test-renderer';
import ProgramEligibility from './ProgramEligibility';

jest.mock(
  'modules/ExpandableContainer',
  () => props => <div className="expandable-container" {...props} />,
);

describe('ProgramEligibility', () => {
  let props;

  beforeAll(() => {
    props = {
      originLatLng: [42.4191639, -71.0269413],
      program: {
        name: 'Program 1',
        receiving_referrals: true,
        status: null,
        email_addresses: ['bobby@gmail.com', 'larry@gmail.com'],
        phone_numbers: [{ country_code: '1', phone_number: '1234442123', phone_type: 'phone' }],
        accessibility_options: [
          {
            id: 'ada_accessible',
            name: 'ADA accessible',
          },
          {
            id: 'translation_services',
            name: 'Interpreter/translation services available',
          },
        ],
        locations: [
          {
            line_1: '123 Loooooooooooooooooooong Street',
            city: 'Funmouth',
            state: 'MA',
            postal_code: '12345',
            country: 'USA',
            address_type: 'work',
            latitude: 42.3682235,
            longitude: -70.2398932,
            is_primary: true,
          },
          {
            line_1: '123 lol Street',
            city: 'Funmouth',
            state: 'MA',
            postal_code: '12345',
            country: 'USA',
            address_type: 'work',
            latitude: 42.3682235,
            longitude: -71.2398932,
            is_primary: true,
          },
        ],
        service_types: [
          {
            id: '50126fcb-a3e9-45d5-9a8e-c32b7fd810b4',
            code: 'UU-BENEFITS',
            name: 'Benefits',
            definition: null,
            created_at: 0,
            updated_at: 0,
            children: [
              {
                id: '2ac1bb8f-89a1-44b9-9e24-9af3cec8a2de',
                code: 'UU-BENEFITS-BENEFITS-ELIGIBILITY-SCREENING',
                name: 'Benefits Eligibility Screening',
                definition: null,
                created_at: 0,
                updated_at: 0,
              },
            ],
            parent_service_type: null,
            facet: null,
          },
        ],
        description: '<div>Hello World</div>',
        eligibility_text: '<p>You are eligible</p>',
      },
      showMore: true,
      showLess: false,
      sectionEmpty: {},
    };
  });

  it('renders ProgramEligibility', () => {
    expect(renderer.create(<ProgramEligibility {...props} />)).toMatchSnapshot();
  });

  it('renders ProgramEligibility eligibility true', () => {
    const newProps = { ...props, eligibility: true };
    expect(renderer.create(<ProgramEligibility {...newProps} />)).toMatchSnapshot();
  });

  it('renders ProgramEligibility geography true', () => {
    const newProps = { ...props, geography: true };
    expect(renderer.create(<ProgramEligibility {...newProps} />)).toMatchSnapshot();
  });
});
