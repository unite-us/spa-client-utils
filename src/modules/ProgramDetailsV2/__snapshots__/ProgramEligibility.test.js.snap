// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramEligibility renders ProgramEligibility 1`] = `
<div
  className="program-description__eligibility"
>
  <div
    className=""
  >
    <div
      className="expandable-container"
      collapsedHeight="70px"
      controlled={true}
      hideToggle={true}
      id="test-expandable"
      isOpen={true}
      showGradient={false}
    >
      <h4
        className="mb-half"
      >
        Eligibility
      </h4>
      <div
        className="ui-form-field ui-draft-editor-field"
      >
        <label
          className="ui-form-field__label sr-only top-0"
          htmlFor="program-eligibility_text"
          style={Object {}}
        >
          Eligibility
        </label>
        <div
          className="html-parsed-text"
        >
          <p>
            You are eligible
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <h4
      className="mb-one"
    >
      Client must live in the following...
    </h4>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <div
      className="label-text-inline mb-one"
    >
      <h4
        className="label-text-inline__label"
      >
        Accessibility
        :
      </h4>
       
      <span>
        ADA accessible, Interpreter/translation services available
      </span>
    </div>
  </div>
</div>
`;

exports[`ProgramEligibility renders ProgramEligibility eligibility true 1`] = `
<div
  className="program-description__eligibility"
>
  <div
    className=""
  >
    <div
      className="expandable-container"
      collapsedHeight="70px"
      controlled={true}
      hideToggle={true}
      id="test-expandable"
      isOpen={true}
      showGradient={false}
    >
      <h4
        className="mb-half"
      >
        Eligibility
      </h4>
      <div
        className="ui-form-field ui-draft-editor-field"
      >
        <label
          className="ui-form-field__label sr-only top-0"
          htmlFor="program-eligibility_text"
          style={Object {}}
        >
          Eligibility
        </label>
        <div
          className="html-parsed-text"
        >
          <p>
            You are eligible
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <h4
      className="mb-one"
    >
      Client must live in the following...
    </h4>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <div
      className="label-text-inline mb-one"
    >
      <h4
        className="label-text-inline__label"
      >
        Accessibility
        :
      </h4>
       
      <span>
        ADA accessible, Interpreter/translation services available
      </span>
    </div>
  </div>
</div>
`;

exports[`ProgramEligibility renders ProgramEligibility geography true 1`] = `
<div
  className="program-description__eligibility"
>
  <div
    className=""
  >
    <div
      className="expandable-container"
      collapsedHeight="70px"
      controlled={true}
      hideToggle={true}
      id="test-expandable"
      isOpen={true}
      showGradient={false}
    >
      <h4
        className="mb-half"
      >
        Eligibility
      </h4>
      <div
        className="ui-form-field ui-draft-editor-field"
      >
        <label
          className="ui-form-field__label sr-only top-0"
          htmlFor="program-eligibility_text"
          style={Object {}}
        >
          Eligibility
        </label>
        <div
          className="html-parsed-text"
        >
          <p>
            You are eligible
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <h4
      className="mb-one"
    >
      Client must live in the following...
    </h4>
  </div>
  <div>
    <hr
      className="ui-divider program-details__divider"
      id=""
      style={Object {}}
    />
    <div
      className="label-text-inline mb-one"
    >
      <h4
        className="label-text-inline__label"
      >
        Accessibility
        :
      </h4>
       
      <span>
        ADA accessible, Interpreter/translation services available
      </span>
    </div>
  </div>
</div>
`;
